Guides/Error-Handling.md

```diff
@@ -14,53 +14,56 @@
 
-**Error Labels:** Some types of errors may contain more specific information describing the context in which they occured. This information is conveyed through the usage of `errorLabels`. Specifically, any server error or connection related error may contain labels.
+**Error Labels:** Some types of errors may contain more specific information describing the context in which they occured. Such errors conform to the `LabeledError` protocol, and convey the extra information is conveyed through the `errorLabels` property. Specifically, any server error or connection related error may contain labels.
 
 ### Server Errors
-Server errors correspond to failures that occur in the database itself and are returned to the driver via some response to a command. Each `ServerError` case contains at least one error code representing what went wrong on the server.
+Server errors correspond to failures that occur in the database itself and are returned to the driver via some response to a command. Each error that conforms to `ServerError` contains at least one error code representing what went wrong on the server.
 
-For an enumeration of the possible server error codes, [see this list](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err).
+For an enumeration of the possible server error codes, [see this list](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml).
 
-The `ServerError` cases are as follows:
-- `.commandError(code: ServerErrorCode, message: String, errorLabels: [String]?)`:
+The possible errors that conform to `ServerError` are as follows:
+- `CommandError`:
     - Thrown when commands experience errors server side that prevent execution.
     - Example command failures include failure to parse, operation aborted by the user, and unexpected errors during execution.
-- `.writeError(writeError: WriteError?, writeConcernError: WriteConcernError?, errorLabels: [String]?)`
-    - Thrown when a single write command fails on the server.
-    - Only one of the two optionals will be non-nil.
-- `.bulkWriteError(writeErrors: [BulkWriteError]?, writeConcernError: WriteConcernError?, result: BulkWriteResult, errorLabels: [String]?)`
+- `WriteError`
+    - Thrown when a single write command fails on the server (e.g. insertOne, updateOne, updateMany).
```

Comment:
should we indicate here what was previously in the docstring about "Only one of the two optionals will be non-nil"?